jms/src/main/scala/akka/stream/alpakka/jms/impl/JmsConnector.scala

```diff
@@ -56,26 +57,40 @@ private[jms] trait JmsConnector[S <: JmsSession] {
   override def preStart(): Unit = {
     // keep two elements since the time between initializing and connected can be very short.
     // always drops the old state, and keeps the most current (two) state(s) in the queue.
-    val pair = Source.queue[InternalConnectionState](2, OverflowStrategy.dropHead).preMaterialize()(this.materializer)
-    connectionStateQueue = pair._1
-    connectionStateSourcePromise.complete(Success(pair._2))
+    val (queue, source) =
+      Source
+        .queue[InternalConnectionState](2, OverflowStrategy.dropHead)
+        .toMat(BroadcastHub.sink(1))(Keep.both)
+        .run()(this.materializer)
+    connectionStateQueue = queue
+    connectionStateSourcePromise.complete(Success(source))
+
+    // add subscription to purge queued connection status events after the configured timeout.
+    val system: ActorSystem = ActorMaterializerHelper.downcast(materializer).system
+    after(jmsSettings.connectionStatusSubscriptionTimeout, system.scheduler) {
+      source.runWith(Sink.ignore)(this.materializer)
+    }
   }
 
   protected def finishStop(): Unit = {
     val update: InternalConnectionState => InternalConnectionState = {
       case JmsConnectorStopping(completion) => JmsConnectorStopped(completion)
-      case current => current
+      case stopped: JmsConnectorStopped => stopped
+      case current =>
+        JmsConnectorStopped(
+          Failure(new IllegalStateException(s"BUG: completing stage stop in unexpected state ${current.getClass}"))
```

Comment:
Sure thing - I wanted to stay in line with this https://github.com/akka/alpakka/blob/63346ea71a8e79669ecab5416f1fe80adadcce2a/jms/src/main/scala/akka/stream/alpakka/jms/impl/JmsConnector.scala#L342 but since it potentially gets shown to clients, let's remove it